test/e2e/autoscale_test.go

```diff
@@ -110,28 +110,60 @@ func tearDown(clients *test.Clients, names test.ResourceNames, logger *zap.Sugar
 	TearDown(clients, names, logger)
 }
 
-func TestAutoscaleUpDownUp(t *testing.T) {
+func TestAutoscaleUpDownUp_WithoutReadinessProbe(t *testing.T) {
 	//add test case specific name to its own logger
-	logger := test.GetContextLogger("TestAutoscaleUpDownUp")
+	logger := test.GetContextLogger("TestAutoscaleUpDownUp_WithoutReadinessProbe")
 
 	clients := setup(t, logger)
-	imagePath := strings.Join(
-		[]string{
-			test.Flags.DockerRepo,
-			"autoscale"},
-		"/")
+	imagePath := strings.Join([]string{test.Flags.DockerRepo, "autoscale"}, "/")
 
 	logger.Infof("Creating a new Route and Configuration")
 	names, err := CreateRouteAndConfig(clients, logger, imagePath)
 	if err != nil {
 		t.Fatalf("Failed to create Route and Configuration: %v", err)
 	}
+
+	test.CleanupOnInterrupt(func() { tearDown(clients, names, logger) }, logger)
+	defer tearDown(clients, names, logger)
+
+	testAutoscaleUpDownUp(t, logger, clients, names)
+}
+
+func TestAutoscaleUpDownUp_WithReadinessProbe(t *testing.T) {
+	//add test case specific name to its own logger
+	logger := test.GetContextLogger("TestAutoscaleUpDownUp_WithReadinessProbe")
+
+	clients := setup(t, logger)
+	imagePath := strings.Join([]string{test.Flags.DockerRepo, "autoscale"}, "/")
+
+	logger.Infof("Creating a new Route and Configuration")
+
+	names := test.ResourceNames{
```

Comment:
Looks like there is test helper function to create route and config `CreateRouteAndConfig`.

Reply:
The `TestAutoscaleUpDownUp_WithReadinessProbe` intentionally avoids the `CreateRouteAndConfig` call, since the generated ConfigurationSpec does not have a readinessProbe. Could have altered the `CreateRouteAndConfig` and passed additional params which would have meant passing it down the chain of calls. Also increased the list of params. Instead opted to go with the approach used by `test/e2e/build_test.go`